CONTRIBUTING.md

```diff
@@ -0,0 +1,101 @@
+# Contributing to Miden
+
+#### First off, thanks for taking the time to contribute!
+
+You can find a detailed explanation of main project concepts in [docs](https://maticnetwork.github.io/miden/).
+
+We want to make contributing to this project as easy and transparent as possible, whether it's:
+
+- Reporting a [bug](https://github.com/maticnetwork/miden/issues/new)
+- Take part in [discussions](https://github.com/maticnetwork/miden/discussions)
+- Submitting a fix
+- Proposing new [features](https://github.com/maticnetwork/miden/issues/new)
+
+&nbsp;
+
+## Flow
+We are using [Github Flow](https://guides.github.com/introduction/flow/index.html), so all code changes happen through pull requests.
+
+### Versioning
+We use [semver](https://semver.org/) naming convention.
+
+### Branching
+- The current active branch is `next`. Every branch with fix/feature must be forked from `next`.
+
+- Branch name should contain a short issue/feature decription separated with hyphens [(kebab-case)](https://en.wikipedia.org/wiki/Letter_case#Kebab_case).
+
+For example, if the issue title is `Fix functionality X in component Y` then branch name will be smth like: `fix-x-in-y`.
```

Comment:
nit: I would probably type out "something" fully here instead of "smth". I would also say "...the branch name..." again instead of just "...branch name..."

Reply:
Done.